### PR TITLE
add a new test case for LRO patch with 200

### PR DESCRIPTION
**Pull request by ArcturusZhang**

### CODE REVIEW

**Comment by deyaaeldeen** [comment]
Could you please having the actual change in a separate commit/PR from the formatting ones?

**Comment by ArcturusZhang** [comment]
Hi @deyaaldeen thank you for your suggestion. I just reverted the format changes, please take a look again

**Review by deyaaeldeen — COMMENTED** [review]
@ArcturusZhang thanks! I see, do we know if there is an azure service that support patching LROs?

@lmazuel it looks good to me, but could you please take a look?

**Comment by ArcturusZhang** [comment]
> @ArcturusZhang thanks! I see, do we know if there is an azure service that support patching LROs?
> 
> @lmazuel it looks good to me, but could you please take a look?

We have a lot of this kind of cases in the compute RP. For instance, the patch for virtual machines: https://github.com/Azure/azure-rest-api-specs/blob/aa9d7789d10539f808419e01833b803849f6d6f6/specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/compute.json#L3175

**Comment by lmazuel** [comment]
I don't think I agree with the concept of the PR, the presence of a Location header implies polling. The Compute Swagger you mention seems wrong to me, if the only acceptable answer is 200, why making it a LRO in the first place? I'm sure there is more to it than the Swagger and the Swagger is not accurate of actual PATCH return. I would be extermely surprised that Compute team is commiting that all PATCH operations on VM are instantaneous.

**Review by lmazuel — CHANGES_REQUESTED** [review]
Feels incorrect

**Comment by ArcturusZhang** [comment]
> I don't think I agree with the concept of the PR, the presence of a Location header implies polling. The Compute Swagger you mention seems wrong to me, if the only acceptable answer is 200, why making it a LRO in the first place? I'm sure there is more to it than the Swagger and the Swagger is not accurate of actual PATCH return. I would be extermely surprised that Compute team is commiting that all PATCH operations on VM are instantaneous.

I am not sure with the compute swagger - we could follow up on that. But after I run the real case again, I found it actually has an `Azure-AsyncOperation` header instead of `Location` header and I have changed that in my PR.

**Review by m-nash — CHANGES_REQUESTED** [review]

File: legacy/routes/lros.js [hunk 1]
```diff
@@ -58,12 +62,30 @@ var lros = function (coverage) {
...
-  coverage['LROPut200InlineCompleteNoState'] = 0;
   router.put('/put/200/succeeded/nostate', function (req, res, next) {
     coverage['LROPut200InlineCompleteNoState']++;
     res.status(200).type('json').end('{"id": "100", "name": "foo" }');
   });
 
+  router.patch("/patch/200/succeeded", function (req, res, next) {
+    coverage["LROPatchInlineComplete"]++;
+    var pollingUri = getRequestBaseUrl(req) + "/lro/patchasync/200/operationResults";
+    var headers = {
```
Comment (lmazuel) [inline legacy/routes/lros.js:73]: To be consistent with the testserver naming convention, the test `LROPatchInlineComplete` of url `/patch/200/succeeded` should not have headers defined, like you can see the parallel in PUT:
https://github.com/Azure/autorest.testserver/blob/ab118a361d2b98697ed4b6ba9450348beeaf6c3d/legacy/routes/lros.js#L55-L58

File: legacy/routes/lros.js [hunk 2]
```diff
@@ -58,12 +62,30 @@ var lros = function (coverage) {
...
+      'Azure-AsyncOperation': pollingUri,
+    };
+    res.set(headers).status(200).type("json")
+      .end('{ "properties": { "provisioningState": "Succeeded"}, "id": "100", "name": "foo" }');
+  });
+
+  router.patch("/patch/200/succeeded/nostate", function (req, res, next) {
```
Comment (lmazuel) [inline legacy/routes/lros.js:80]: You may just waive this improbably never seen scenario for now, or if you keep it use the PUT naming convention too:
https://github.com/Azure/autorest.testserver/blob/ab118a361d2b98697ed4b6ba9450348beeaf6c3d/legacy/routes/lros.js#L65-L68

**Comment by lmazuel** [comment]
@ArcturusZhang @m-nash to be faithful of what you want, please do a new test called something like:

```
  router.patch("/patch/200/succeeded/ignoreheaders", function (req, res, next) {
    coverage["LROPatchInlineCompleteIgnoreHeaders"]++;
    var headers = {
      'Azure-AsyncOperation': "http://localhost/fakeurl_crash_with_404_you_should_not_call_this",
    };
    res.set(headers).status(200).type("json")
      .end('{ "properties": { "provisioningState": "Succeeded"}, "id": "100", "name": "foo" }');
  });
```

This way you make the point of the test more obvious: runtime should ignore headers if the situation is 200/Succeeded. If they don't this will crash since the polling will raise from a 404.

**Review by lmazuel — APPROVED** [review]

**Comment by lmazuel** [comment]
@ArcturusZhang you need to update the bugfix version number in your PR (that's the way the CI is setup) from 31 to 32. On merge, the version 32 will get released automatically. But if you don't increment the number, you won't get a release.

https://github.com/Azure/autorest.testserver/blob/2d144ee8c1b5632b61f479650c2b7e6ef816bc55/package.json#L3

**Review by m-nash — APPROVED** [review]

**Pull request merged**